src/m365/spo/commands/serviceprincipal/serviceprincipal-permissionrequest-list.ts

```diff
@@ -22,6 +23,11 @@ class SpoServicePrincipalPermissionRequestListCommand extends SpoCommand {
   public async commandAction(logger: Logger): Promise<void> {
     try {
       const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
+      const spoClientExtensibilityWebApplicationPrincipalId = await this.getSPOClientExtensibilityWebApplicationPrincipalId();
```

Comment:
From performance point of view, wouldn't it be better to move these calls only after we retrieved permission requests, the call wasn't erroneous and we have at least one request? Right now, it seems like we might be doing all this extra work for nothing, if there are no pending requests or if retrieving them will fail with an error. Let's move these calls after `const result: SPOWebAppServicePrincipalPermissionRequest[] = json[json.length - 1]._Child_Items_;` and after we checked that the array isn't empty.